### DISCUSSION TITLE
Is there a way to list all notes/todos from all subfolders, or all notebooks?

### DISCUSSION THREAD

**Discussion by Callumk7** [xwmx/nb#299] [Q&A]
As far as I can see, there is no way to list ALL of a notebook's content using nb ls, it just shows the root folder. Is it possible to list the entire contents of a notebook?

**Comment by xwmx** [comment]
There’s an undocumented `—-tree` / `—-recursive` option for `nb`, `nb ls`, and `nb list` that does this. Some features don’t work with this option yet, such as filtering.

**Reply by theutz** [reply]
This flag doesn't seem to work for me at all, unfortunately.

**Reply by theutz** [reply]
Never mind. After looking through the tests, it appears it works if you specify the notebook with a selector like so `nb ls home: --recursive` or `nb ls home: --tree`.

**Reply by neongreen-sc** [reply]
still doesn't work for me 🤔 does it have to be enabled somehow?

**Reply by cbrasser** [reply]
That does not work for me either. I'd be interested in a solution for this Problem.

**Reply by nobodyinperson** [reply]
Same, this makes folders a bit useless 😅

EDIT: Ah, adding a trailing slash to a folder name at least lists its contents:

```bash
nb ls folder/
```

**Reply by jupart** [reply]
Having this work for tasks as well would be very helpful to my workflow. Right now all my files are in the root notebook, daily log files and individual project files, ideas, etc. It'd be awesome to be able to organize them into folders, which of course I can. But if I drop a task in one of those files (like I can do with the root approach), it won't show unless I write some tool to iterate over folders.

**Reply by xwmx** [reply]
There are [tests for this in `todo-tasks.bats`](https://github.com/xwmx/nb/blob/25aa598ca0f898da23cd3bda8a69f5225eba78d7/test/todo-tasks.bats#L6) passing on macOS and Ubuntu.

If someone who is encountering this issue can clone the repo and run `bats test/todo-tasks.bats` to see if there are any errors, that could help us identify what’s happening.

**Reply by jupart** [reply]
Got `36 tests, 0 failures` doing this.

**Reply by jupart** [reply]
I ran `nb update` just in case to update my package manager installed `nb`, have a folder with a test file and a test task in the file (just a heading and task) and `nb tasks --recursive` does not pick it up.

**Reply by jupart** [reply]
```
~ took 11s
❯ nb test:tasks --recursive
! 0 tasks.

~
❯ ls -T ~/.nb/test
/home/jupart/.nb/test
└── projects
    └── 20250708150800.md

~ took 5s
❯ nb test:show projects/1 -p
# Test item

- [ ] And task

```
